pkg/cluster/delete.go

```diff
@@ -356,6 +358,73 @@ func (m *manager) deleteGateway(ctx context.Context) error {
 	return nil
 }
 
+func (m *manager) deleteClusterMsiCertificate(ctx context.Context) error {
+	// The cluster MSI may have been deleted prior to cluster deletion. If that's the case
+	// we will have already deleted the certificate.
+	if m.doc.OpenShiftCluster.Identity == nil || m.doc.OpenShiftCluster.Identity.UserAssignedIdentities == nil || len(m.doc.OpenShiftCluster.Identity.UserAssignedIdentities) == 0 {
+		m.log.Warning("skipping cluster MSI certificate deletion because cluster MSI has already been deleted")
+		return nil
+	}
+
+	secretName, err := m.clusterMsiSecretName()
+	if err != nil {
+		return err
+	}
+
+	err = m.clusterMsiKeyVaultStore.DeleteCredentialsObject(ctx, secretName)
+	if err == nil || azuresdkerrors.IsNotFoundError(err) {
+		return nil
+	}
+
+	return err
+}
+
+func (m *manager) deleteFederatedCredentials(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile == nil {
+		return nil
+	}
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		identityResourceId, err := azure.ParseResourceID(identity.ResourceID)
+		if err != nil {
+			return err
+		}
+
+		federatedIdentityCredentialResourceName, err := m.getPlatformWorkloadIdentityFedertedCredName(identity)
+		if err != nil {
+			return err
+		}
+
+		_, err = m.clusterMsiFederatedIdentityCredentials.Get(
```

Comment:
Similar to creation of federated credentials, only get and delete the federated credentials present in `platformWIRolesByRoleName`.

Reply:
updated